xbmc/utils/log.cpp

```diff
@@ -85,11 +85,14 @@ void CLog::LogFunction(int loglevel, const char* functionName, const char* forma
 
 void CLog::LogFunction(int loglevel, const char* functionName, int component, const char* format, ...)
 {
-  if (g_advancedSettings.CanLogComponent(component))
+  if (g_advancedSettings.CanLogComponent(component) && IsLogLevelLogged(loglevel))
   {
+    std::string fNameStr;
+    if (functionName && functionName[0])
+      fNameStr.assign(functionName).append(": ");
     va_list va;
     va_start(va, format);
-    CLog::LogFunction(loglevel, functionName, format, va);
+    LogString(loglevel, fNameStr + StringUtils::FormatV(format, va));
```

Comment:
Can't you call
```
CLog::LogFunction(loglevel, functionName, StringUtils::FormatV(format, va));
```
That way you don't have to duplicate the function name formatting logic.